shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java

```diff
@@ -244,4 +248,31 @@ public AuthorizationPolicy fetchOrCreateBridgeAuthorizationPolicy(ManagedBridge
 
         return existing;
     }
+
+    @Override
+    public boolean isBridgeStatusChange(ManagedBridge updatedBridge) {
```

Comment:
I believe we should probably also include whether there are _new_ `Condition`'s or _removed_ `Condition`'s. This is not strictly necessary when the `Set` of `Condition`'s remains unchanged however given [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1438/files#r1051955973) discussion that can no longer be assumed to be true.

Reply:
sorry, I didn't get you. Could you please explain

Reply:
OK, sure. 

So if the original set of `Condition`'s for _create_ contains [`Create1=Unknown`, `Create2=True`] which then becomes [`Create1=True` `Create2=True`] your logic detects this has changed. Great. When we then delete the resource the set of `Conditon`'s for _delete_ contains [`Delete1=Unknown`, `Delete2=Unknown`] your logic would fail to detect this change. In fact I suspect your [code](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1438/files/468203f50e91160441a47cda31801438e84d7787#diff-c352b575f3a8abde0ace3a82d0f8e0e87a5398d47aea81233053bf7b6d961775R271) would throw a `NullPointerExcepton` as the `updatedCondition` would be `null` (as the _old name_ `Create1` cannot be found in the new set of [`Delete1`, `Delete2`]).

Reply:
- there would not be any `Delete1=Unknown, Delete2=Unknown`. There is only one delete `BridgeDeleted=True`.
- `BridgeDeleted` would not be maintain inside CRD status. Deleted CRDs would be identify by the `sync with manager` part of operator and directly send back to manager through DTO if some CRDs don't exists.
- `public boolean isBridgeStatusChange(ManagedBridge updatedBridge)` would always works with fixed set of conditions.